lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -346,9 +348,34 @@ def _on_start(self, event: StartEvent):
         # configure clients auth
         self.opensearch_config.set_client_auth()
 
+        deployment_desc = self.opensearch_peer_cm.deployment_desc()
+        # only start the main orchestrator if a data node is available
+        # this allows for "cluster-manager-only" nodes in large deployments
+        # workflow documentation:
+        # no "data" role in deployment desc -> start gets deferred
+        # when "data" node joins -> start cluster-manager via _on_peer_cluster_relation_changed
+        # cluster-manager notifies "data" node via refresh of peer cluster relation data
+        # "data" node starts and initializes security index
+        if (
+            deployment_desc.typ == DeploymentType.MAIN_ORCHESTRATOR
+            and not deployment_desc.start == StartMode.WITH_GENERATED_ROLES
+            and "data" not in deployment_desc.config.roles
+        ):
+            self.status.set(BlockedStatus(PClusterNoDataNode))
+            event.defer()
+            return
+
         # request the start of OpenSearch
         self.status.set(WaitingStatus(RequestUnitServiceOps.format("start")))
-        self._start_opensearch_event.emit()
+
+        # if this is the first data node to join, start without getting the lock
+        ignore_lock = (
```

Comment:
Hey Pedro, thanks for your comment. I've been thinking about it a bit, and I don't think that the 2nd case (node lock not available) is an issue. If this happens, because no data node is online, the API-calls will not be successful, and in that case, the operator should fall back to the peer database lock (see [here](https://github.com/canonical/opensearch-operator/blob/main/lib/charms/opensearch/v0/opensearch_locking.py#L279-L287)).

Reply:
@reneradoi well, the snippet you showed means we can have up to 2x nodes running (1st booting is the CM unit leader, 2nd is the next unit that tries to get the lock and sees a single CM up).

However, I cannot think of a scenario where this may be a problem right now... Only case that concerns me is if we have the entire data application powered down. But only case I can think of is a restart of the app itself. I will close this topic and I think we should focus instead on: https://github.com/canonical/opensearch-operator/pull/424#pullrequestreview-2302418430